### PR TITLE
complete refactor of model objects

### PR DESCRIPTION
**Pull request by jeremymanning**
so many things have changed...need to go through individual files to see everything
- changed numerators and denominators to be represented via logs
- changed init and update for model objects
- a bunch of efficiency changes
- consolidated "three cases" of timeseries recon into a single function
- probably other stuff

### CODE REVIEW

File: supereeg/helpers.py [hunk 1]
```diff
@@ -506,27 +497,48 @@ def _timeseries_recon(bo, K, chunk_size=1000, preprocess='zscore'):
...
+    if ~np.any(brain_locs_in_model):
+        #if none of the brain locations are in the model, we need to blur out the model to match up with the
+        # locations in the brain object
+        combined_locs = np.vstack((bo.get_locs(), mo.get_locs()))
+        model_locs_in_brain = [False]*bo.get_locs().shape[0]
+        model_locs_in_brain.extend([True]*mo.get_locs().shape[0])
+
+        rbf_weights = _log_rbf(combined_locs, mo.get_locs())
+        Z = _blur_corrmat(Z, rbf_weights)
+
```
Comment (jeremymanning) [inline supereeg/helpers.py]: Observation from @lucywowen: what if the brain locations *partially* overlap with the model locations?

Comment (jeremymanning) [reply supereeg/helpers.py]:
```
if ~np.all(brain_locs_in_model):
    combined_locs = np.vstack((bo.get_locs(), mo.get_locs()))
    if np.any(brain_locs_in_model):
        Combined_locs = np.unique_rows(combined_locs)
        < update model_locs_in_brain>
    Else:
        Model_locs_in_brain = [False]*bo.get_locs().shape[0]
        Model_locs_in_brain.extend([True]*mo.get_locs().shape[0])
    Rbf_weights = ...
    Z = ...

**Comment by andrewheusser** [comment]
all tests are go!

**Comment by jeremymanning** [comment]
Are we ready to merge then?

**Pull request merged**